tests/integration-tests/tests/integration/name_server_pool_tests.rs

```diff
@@ -665,7 +685,11 @@ fn test_user_provided_server_order() {
     let pool = mock_nameserver_pool(
         vec![preferred_nameserver, secondary_nameserver],
         None,
-        options,
+        NameServerOptions {
```

Comment:
Assuming this syntax still works when `NameServerOptions` becomes non-exhaustive, suggest an intermediate commit that transitions to the partial update syntax in a separate commit from the one changing from `ResolverOpts` to `NameServerOptions`.